src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -122,6 +161,39 @@ export class PokemonSaveParser {
     }
   }
 
+  /**
+   * Initialize memory mode with WebSocket client and auto-detect config
+   */
+  private async initializeMemoryMode (client: MgbaWebSocketClient): Promise<void> {
+    this.webSocketClient = client
+    this.isMemoryMode = true
+
+    // Check if connected
+    if (!client.isConnected()) {
+      throw new Error('WebSocket client is not connected to mGBA')
+    }
+
+    // Get game title to check compatibility
+    const gameTitle = await client.getGameTitle()
+    if (!this.silent) console.log(`Memory mode: Connected to game "${gameTitle}"`)
+
+    // Auto-detect config based on game title using proper config system
+    if (!this.config) {
+      this.config = GameConfigRegistry.getConfigForMemory(gameTitle)
+
+      if (!this.config) {
+        throw new Error(`Unsupported game for memory parsing: "${gameTitle}". No compatible config found.`)
+      }
+    } else {
+      // Verify existing config can handle this game
+      if (!this.config.canHandleMemory?.(gameTitle)) {
+        throw new Error(`Current config "${this.config.name}" cannot handle memory parsing for "${gameTitle}"`)
+      }
+    }
+
+    if (!this.silent) console.log(`Memory mode initialized for ${gameTitle} using config: ${this.config.name}`)
+  }
```

Comment:
is there a better way we can more neatly with the savefile parsing init this stuff?